nupic/utils.py

```diff
@@ -114,3 +125,48 @@ def write(self, proto):
     proto.windowSize = self.windowSize
     proto.slidingWindow = self.slidingWindow
     proto.total = self.total
+
+###########################################################################
+class CircularBuffer(object):
+  """
+  implementation of a fixed size constant random access circular buffer
+  #TODO can this be merged with MovingAverage somehow?
```

Comment:
Yeah I don't know if we should keep it or not. I'd just rather not complicate this PR with an unrelated task of moving this class around.